### ISSUE TITLE
CSV data load/save in frontend member modules

### ISSUE THREAD

**Issue by qzminski** [contao/contao#4980]
### Affected version(s)

4.9+

### Description

I have some custom member fields that have the `'eval' => ['csv' => ',']` setting. Unfortunately, loading and saving the data does not work correctly in the frontend member modules such as registration or personal data modules.

In the first place, I thought of creating a simple pull request to add this functionality directly in the modules, similar to what [data containers do](https://github.com/contao/contao/blob/4.13/core-bundle/src/Resources/contao/drivers/DC_Table.php#L3208-L3212). However, this would undoubtedly lead to a duplicate code problem. We should centralize the IO data handling at some point, but I am not entirely sure if it'd be a bug fix or a completely new feature.

Please let me know whether I should provide a pull request that fixes this problem in the simplest possible way, or we should discuss a new concept to unify it all.

**Comment by aschempp** [comment]
Since we (currently) re-implement everything in these modules, I think we need to also do that in this case. Can you give a real-world example of serialized data that can be entered in the front end? For example the back end text field supports "multiple" input fields where this might make sense, but the frontend form field does not?

**Comment by qzminski** [comment]
Sorry, I was not precise enough. The problem is mainly about the multiple checkboxes. There's likely no field in the core that is affected, but here is one example from my project:

```php
$GLOBALS['TL_DCA']['tl_member']['fields']['app_occupationalGroup'] = [
    'exclude' => true,
    'filter' => true,
    'inputType' => 'checkbox',
    'options' => ['jurist', 'merchant', 'expert'],
    'reference' => &$GLOBALS['TL_LANG']['APP']['occupationalGroupRef'],
    'eval' => ['multiple' => true, 'csv' => ',', 'feEditable' => true, 'feViewable' => true, 'feGroup' => 'personal', 'tl_class' => 'clr'],
    'sql' => ['type' => 'string', 'length' => 191, 'default' => ''],
    'load_callback' => [
        [\App\EventListener\MemberListener::class, 'onCsvFieldLoadCallback'],
    ],
    'save_callback' => [
        [\App\EventListener\MemberListener::class, 'onCsvFieldSaveCallback'],
    ],
];
```

As you can see, I had to add some custom load and save callbacks for the frontend to parse it correctly:

```php
/**
 * On CSV field load callback.
 */
public function onCsvFieldLoadCallback($value, $source): array
{
    if ($source instanceof FrontendUser) {
        $value = StringUtil::trimsplit(',', $value);
    }

    return $value;
}

/**
 * On CSV field save callback.
 */
public function onCsvFieldSaveCallback($value, $source): string
{
    if ($source instanceof FrontendUser) {
        $value = implode(',', (array) StringUtil::deserialize($value));
    }

    return $value;
}
```

Otherwise, it would save the data as a serialized array, and in the backend, it would look like this:

```
- [ ] Jurist 
- [ ] Merchant
- [ ] Expert
- [x] Undefined option: a:2:{i:0;s:6:"jurist";i:1;s:8:"merchant";}
```

**Comment by qzminski** [comment]
How do we continue from here?

**Comment by leofeyer** [comment]
I cannot reproduce the issue in Contao 4.9. Here is what I have done:

1. I have added `'csv' => ','` to the `tl_member.groups` field in the DCA.
2. I have assigned groups to and saved the members of the Contao demo.
3. The values are loaded and saved correctly.

<img width="414" alt="" src="https://user-images.githubusercontent.com/1192057/192738417-725d7b10-9e21-41db-b7d8-6529d45f5c02.png">

<img width="284" alt="" src="https://user-images.githubusercontent.com/1192057/192738524-850c8451-debc-4896-aa3b-2a19e4b36f23.png">

I assume I have to do something different to reproduce the issue?
